### PR TITLE
framework/task_manager: implement task manager restart API

### PR DESCRIPTION
**Pull request by jeongarmy**
The restart API supports functionality to restart running task.

### CODE REVIEW

File: framework/src/task_manager/task_manager_core.c [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#include <sched.h>
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_core.c]: For what?

File: apps/examples/task_manager_sample/user.c [hunk 2]
```diff
@@ -82,9 +84,18 @@ int user_main(int argc, char *argv[])
...
 	}
 
-	sleep(2);
-	printf("\nUser App sends Alarm Off message to Action Manager\n");
+	sleep(3);
+	printf("\nUser App sends Alarm Restart message to Action Manager\n");
 	ret = task_manager_unicast(handle_actionmanager, msg4, strlen(msg4) + 1, TM_RESPONSE_WAIT_INF);
+	if (ret != OK) {
+		printf("ERROR : SEND SIGNAL %d\n", ret);
+	} else if (ret == OK) {
```
Comment (jeongchanKim) [inline apps/examples/task_manager_sample/user.c]: only else is enough

File: framework/src/task_manager/task_manager_restart.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <string.h>
+#include <stdio.h>
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_restart.c]: Please check whether all headers are used or not.

File: framework/include/task_manager/task_manager.h [hunk 4]
```diff
@@ -178,6 +179,18 @@ int task_manager_pause(int handle, int timeout);
  * @since TizenRT v2.0 PRE
  */
 int task_manager_resume(int handle, int timeout);
+/**
+ * @brief Request to restart the task
```
Comment (jeongchanKim) [inline framework/include/task_manager/task_manager.h]: please add description that restart cannot guarantee the memory deallocation

**Comment by seinfra** [comment]
Target : [2787ebdb16774ff8237eb01bd9f22e2f79ada52d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2787ebdb16774ff8237eb01bd9f22e2f79ada52d] - Code Rule Check OK.

File: apps/examples/task_manager_sample/user.c (hunk 2 above)
Comment (jeongarmy) [reply apps/examples/task_manager_sample/user.c]: will modify all similar codes later

File: framework/include/task_manager/task_manager.h (hunk 4 above)
Comment (jeongarmy) [reply framework/include/task_manager/task_manager.h]: Done

**Comment by seinfra** [comment]
Target : [b770f015c2e842861ad3747f51f70b8344f3d62b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b770f015c2e842861ad3747f51f70b8344f3d62b] - Code Rule Check (C++) OK.

File: framework/src/task_manager/task_manager_core.c (hunk 1 above)
Comment (jeongarmy) [reply framework/src/task_manager/task_manager_core.c]: For task_restart

File: framework/src/task_manager/task_manager_restart.c (hunk 3 above)
Comment (jeongarmy) [reply framework/src/task_manager/task_manager_restart.c]: Yes. all headers are needed.
string.h for memset, stdio.h for asprintf, unistd.h for getpid, sys/types.h for OK

**Comment by seinfra** [comment]
Target : [675a3451859910afcce90b888b11c6a6c9d34e6d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [675a3451859910afcce90b888b11c6a6c9d34e6d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ae2024d5a16fcc67bb93ca5a74096b7ca178317a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ae2024d5a16fcc67bb93ca5a74096b7ca178317a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c0eb858535f4cb7740197d07c7ae07007139c419] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c0eb858535f4cb7740197d07c7ae07007139c419] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7aa0f85b5d04c9c8c32fc9bdf2cb6189861b3287] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7aa0f85b5d04c9c8c32fc9bdf2cb6189861b3287] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b653f681718997ccf4654b8131ac97d4ec48aab3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b653f681718997ccf4654b8131ac97d4ec48aab3] - Code Rule Check OK.

**Pull request merged**